core/src/modules/configLoaders/index.ts

```diff
@@ -87,9 +88,38 @@ async function loadConfigFile(file: string): Promise<ConfigurationObject> {
   return payload;
 }
 
+export async function shouldExternallyValidate(
+  canExternallyValidate: boolean,
+  configObject: ConfigurationObject,
+  configObjects: ConfigurationObject[],
+  locallyValidateIds: Set<string>
+) {
+  if (!canExternallyValidate) return false;
+  if (!locallyValidateIds) return true;
+
+  const { prerequisiteIds: parentIds } = await getParentIds(
+    configObject,
+    configObjects
+  );
+
+  const objectId = configObject.id;
+
+  const localParents = parentIds.filter((pId) => locallyValidateIds.has(pId));
+  if (localParents.length > 0) {
+    locallyValidateIds.add(objectId);
+  }
+
+  if (locallyValidateIds.has(objectId)) {
+    return false;
+  }
+
+  return true;
+}
+
 export async function processConfigObjects(
   configObjects: Array<ConfigurationObject>,
-  externallyValidate: boolean,
+  canExternallyValidate: boolean,
+  locallyValidateIds?: Set<string>,
```

Comment:
Comment (no need to change it): This might be the first use of `Set` in the codebase!  Given the rest of the code, is this a surprising data structure, or since we have TS helping us out, is this OK?  Set is certainly the best, being that we don't want duplicates.

Reply:
We've used it a few times in some of the plugins, but yeah I dont think it's been used in core. Since we're adding the children IDs I wanted to avoid duplicates. An alternative is also to take in an array like usual and make the set within the funciton